Docker image

philpep opened the pull request.

philpep committed:
Drop support of centos6 in tests

philpep committed:
Use a single Dockerfile

Merge centos7 and centos7_salted by using a build arg variable
containing the state to run (state.highstate by default).

Having a single dockerfile simplify the config for building image on the
docker hub.

philpep committed:
Do not start systemd as PID 1 in docker

Using systemd in docker require the --privileged flag which can be
avoided since we don't rely on systemd to run supervisor.

Add a docker entrypoint that run supervisord as ckan user instead.
For "./run-tests.py dev" run a endless command (tail -F /dev/null) since
supervisor is not deployed.

Make ./run-tests.py --cmd exec commands using root user like before (so
ignore USER in Dockerfile)

philpep committed:
Start redis using supervisor

Since redis don't start by itself in docker

philpep committed:
Install all multilingual dataset in solr

Otherwise solr won't start.

philpep committed:
Use latest virtualenv instead of latest pip

Since a latest virtualenv will deploy a latest pip.
Also this will install a new version of setuptools to avoid
"AttributeError: 'EntryPoint' object has no attribute 'resolve'" at ckan
start.

dlax commented on ckan/install.sls:
```diff
@@ -35,28 +35,25 @@ ckan-user:
+  cmd.run:
+    - name: pip install --user virtualenv
+    - user: {{ ckan.ckan_user }}
+    - creates: {{ home }}/.local/bin/virtualenv
```
use `pip.installed` and pass `upgrade: true`

philpep replied:
Unfortunately, seems that salt pip.installed has no equivalent for `--user`.

https://docs.saltstack.com/en/latest/ref/states/all/salt.states.pip_state.html#module-salt.states.pip_state

dlax reviewed (changes requested):
As a follow-up to Solr fixes, you can look at https://hub.docker.com/r/ckan/solr/~/dockerfile/

dlax commented:
Philippe Pepiot a écrit :
> *@philpep* commented on this pull request.
> 
> ------------------------------------------------------------------------
> 
> In ckan/install.sls 
> <https://github.com/logilab/ckan-formula/pull/11#discussion_r148781489>:
> 
>>        - {{ ckan.python_pip }}
>   
> +  cmd.run:
> +    - name: pip install --user virtualenv
> +    - user: {{ ckan.ckan_user }}
> +    - creates: {{ home }}/.local/bin/virtualenv
> 
> Unfortunately, seems that salt pip.installed has no equivalent for |--user|.
> 
> https://docs.saltstack.com/en/latest/ref/states/all/salt.states.pip_state.html#module-salt.states.pip_state
> 

You can use:

   - install_options:
     - --user

That's already used in ckan/supervisor.sls.

philpep commented:
On 03/11/2017, Denis Laxalde wrote:
> Philippe Pepiot a écrit :
> > *@philpep* commented on this pull request.
> > 
> > ------------------------------------------------------------------------
> > 
> > In ckan/install.sls 
> > <https://github.com/logilab/ckan-formula/pull/11#discussion_r148781489>:
> > 
> >>        - {{ ckan.python_pip }}
> >   
> > +  cmd.run:
> > +    - name: pip install --user virtualenv
> > +    - user: {{ ckan.ckan_user }}
> > +    - creates: {{ home }}/.local/bin/virtualenv
> > 
> > Unfortunately, seems that salt pip.installed has no equivalent for |--user|.
> > 
> > https://docs.saltstack.com/en/latest/ref/states/all/salt.states.pip_state.html#module-salt.states.pip_state
> > 
> 
> You can use:
> 
>    - install_options:
>      - --user
> 
> That's already used in ckan/supervisor.sls.

Didn't known that --user was a setup.py install option. Great, thanks!

philpep committed:
fixup install pip virtualenv

philpep committed:
Add README about docker image usage

philpep commented:
updated / rebased the PR and added some usage of the docker image in the README

philpep commented:
Updated to fix travis tests that run `./run-tests.py dev --cmd "salt-call ..."` where salt-call was called as ckan user (USER in Dockerfile) instead of root. Fixed by running command passed in `--cmd` as root user.
